specification/dsdl/serialization.tex

```diff
@@ -77,12 +82,81 @@ \subsubsection{Implicit padding}
+The specifics of such padding behavior fall outside of the scope of the DSDL specification
+as they belong to the domain of network transports rather than data presentation.
+
+Another consequence is that a serialized representation constructed from an instance of type $B$ can be
+deserialized into an instance of type $A$ as long as $B$ is a structural subtype of $A$.
+The topic of data type compatibility is explored in section \ref{sec:dsdl_versioning}.
+
+\begin{remark}
+    Let $X$ be an instance of data type $B$, which is defined as follows:
+
+    \begin{minted}{python}
+        float32 parameter
+        float32 variance
+    \end{minted}
+
+    Let $A$ be a structural supertype of $B$, being defined as follows:
+
+    \begin{minted}{python}
+        float32 parameter
+    \end{minted}
+
+    Then the serialized representation of $X$ can be deserialized into an instance of $A$.
+
+    The implicit truncation rule is introduced to permit extension of data types by introducing additional fields
+    without breaking backward compatibility with existing deployments,
+    and to permit introduction of more specific derived types while retaining compatibility with their base types.
+    The topic of data type compatibility is explored in detail in section \ref{sec:dsdl_versioning}.
+\end{remark}
 
-When a serialized representation is deconstructed, the value of the trailing padding bits must be ignored.
+\subsubsection{Implicit zero extension of missing data}
+
+For the purposes of deserialization routines,
+the serialized representation of any instance of any data type shall be considered to contain an implicit
+sequence of trailing zero (0) bits of infinite length.
+
+Despite this rule, implementations are not allowed to intentionally truncate trailing zeros
+upon construction of a serialized representation of an object\footnote{%
+    Intentional truncation is prohibited because in a future revision of the specification the implicit zero
+    extension rule may be removed.
+    If intentional truncation were allowed, removal of the implicit zero extension rule would have
+    broken backward compatibility.
+}.
+
+\begin{remark}
+    This example assumes that the reader is familiar with the variable-length array serialization rules,
+    explained later in this section.
+
+    Let the data type $A$ be defined as follows:
+
+    \begin{minted}{python}
+        uint8 scalar
+    \end{minted}
+
+    Let $X$ be an instance of $A$, where the value of \verb|scalar| is 4.
+    Let the data type $B$ be defined as follows:
+
+    \begin{minted}{python}
+        uint8[<256] array
+    \end{minted}
+
+    Then the serialized representation of $X$ can be deserialized into an instance of $B$ where the field
+    \verb|array| contains a sequence of four zeros: $0, 0, 0, 0$.
+
+    The implicit truncation rule is introduced to permit extension of data types by introducing additional fields
```

Comment:
This paragraph seems misplaced. Should it appear at the start of this section?

Reply:
Sorry, this got messed up. Fixed.